ex04.ex

```diff
@@ -54,8 +58,28 @@ defmodule Ex04 do
       -7
 
   """
+  #def mins([ h | t ]), do: mins( t, h )
+  #def mins([ h | t ], state) do 
+  #case h < state do
+  #true -> mins( t, h )
+  #false -> mins( t, state )
+  #end
+  #end
+  #def mins([], state), do: state
+
+
+
+  def min([ h | t ]) do
```

Comment:
## -2

This is basically an `if` statement, which I'd like you to avoid:

~~~ elixir
  def min(list), do: reduce(list, &min_of/2)
  def min_of(a,  b) when a < b, do: a
  def min_of(_a, b), do: b
~~~